src/Network/Network/Models/PSApplicationGatewayListener.cs

```diff
@@ -31,6 +31,8 @@ public class PSApplicationGatewayListener : PSChildResource
         public string Type { get; set; }
         public PSResourceId SslProfile { get; set; }
 
+        public List<string> HostNames { get; set; }
```

Comment:
HostName

Reply:
The parameter HostNames in ApplicationGatewayListener has been used to maintain parity with the parameter
 Hostnames in ApplicationGatewayHttpListener.
 https://docs.microsoft.com/en-us/powershell/module/az.network/add-azapplicationgatewayrequestroutingrule?view=azps-7.1.0. 
 All the downstream NRP code for this has  been implemented following the same motivation 
 and draws inspiration from existing naming conventions.

Reply:
AzureApplicationGatewayHttpListenerBase has another parameter called HostName, I don't think this is the same case

Reply:
Purpose of the Hostnames filed is same in both AzureApplicationGatewayHttpListenerBase  and AzureApplicationGatewayListenerBase , both can hold multiple string values, Only diffrence is AzureApplicationGatewayHttpListenerBase  is for Layer 7 and AzureApplicationGatewayListenerBase  is for Layer 4 protocol.

Since Hostnames is an array of strings , that might have been the reason why it is named so in AzureApplicationGatewayHttpListenerBase   at first place and we would want to go with the same naming convention for ApplicationGatewayListener as well to maintain parity with already existing Layer 7 cmdlets

Reply:
Use singular noun for PowerShell cmdlet parameter name, there is no need to discuss it.
The reason why there are "HostNames" is because there is another property "HostName". Like I said, it's bad naming in the first place, 
"HostNames" and "HostName" in "AzureApplicationGatewayHttpListenerBase" are not good parameter names in the first place
![Screenshot 2023-10-24 134150](https://github.com/Azure/azure-powershell/assets/19599311/bb602974-1338-4765-93b8-462c8586a879)
I suggest to have alias for these two parameters (make them more meaningful, hostname and hostnames together don't make any sense) in "AzureApplicationGatewayHttpListenerBase". And then, use "HostName" or whatever make more sense in this PR.